Hercules-DJControl-Inpulse-200-script.js

```diff
@@ -0,0 +1,218 @@
+    if (value) {
+        if (DJCi200.scratchButtonStateDA) {
+            DJCi200.scratchButtonStateDA = false;
+            midi.sendShortMsg(0x91, 0x03, 0x00);
+
+        } else {
+            DJCi200.scratchButtonStateDA = true;
+            midi.sendShortMsg(0x91, 0x03, 0x7F);
+
+        }
+    }
+};
+
+DJCi200.vinylButtonDB = function(channel, control, value, status, group) {
+    if (value) {
+        if (DJCi200.scratchButtonStateDB) {
+            DJCi200.scratchButtonStateDB = false;
+            midi.sendShortMsg(0x92, 0x03, 0x00);
+
+        } else {
+            DJCi200.scratchButtonStateDB = true;
+            midi.sendShortMsg(0x92, 0x03, 0x7F);
+
+        }
+    }
+};
+
+// The touch action over the jog wheel
+
+DJCi200.wheelTouchDA = function(channel, control, value, status, group) {
+    channel = channel + 1;
+    if (value > 0 && (engine.getValue("[Channel1]", "play") !== 1 || DJCi200.scratchButtonStateDA)) {
+        //  Touching the wheel.
+        var alpha = 1.0 / 8;
+        var beta = alpha / 32;
+        engine.scratchEnable(1, 400, 33 + 1 / 3, alpha, beta);
+    } else {
+        // Released the wheel.
+        engine.scratchDisable(1);
+    }
+};
+
+DJCi200.wheelTouchDB = function(channel, control, value, status, group) {
+    channel = channel + 2;
+    if (value > 0 && (engine.getValue("[Channel2]", "play") !== 1 || DJCi200.scratchButtonStateDB)) {
+        // Touching the wheel.
+        var alpha = 1.0 / 8;
+        var beta = alpha / 32;
+        engine.scratchEnable(2, 400, 33 + 1 / 3, alpha, beta);
+    } else {
+        // Released the wheel.
+        engine.scratchDisable(2);
+    }
+};
+
+
+
+// Using the top of wheel for scratching (Vinyl button On) and bending (Vinyl button Off)
+DJCi200.scratchWheelDA = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    if (engine.isScratching(1)) {
+        engine.scratchTick(1, newValue); // Scratch!
+    } else {
+        engine.setValue('[Channel' + 1 + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+DJCi200.scratchWheelDB = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    // In either case, register the movement
+    if (engine.isScratching(2)) {
+        engine.scratchTick(2, newValue); // Scratch!
+    } else {
+        engine.setValue('[Channel' + 2 + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+// Using the side of wheel for the bending
+DJCi200.bendWheelDA = function(channel, control, value, status, group) {
+
+    var newValue;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+
+    {
+        engine.setValue('[Channel' + 1 + ']', 'jog', newValue); // Pitch bend
+    }

```

Comment:
Why the curly brackets? Is there supposed to be some if condition that is missing?